src/iuse.cpp

```diff
@@ -3524,14 +3524,10 @@ int iuse::can_goo( player *p, item *it, bool, const tripoint & )
             tries++;
         } while( g->m.impassable( goop ) &&
                  g->m.tr_at( goop ).is_null() && tries < 10 );
-        if( tries < 10 ) {
```

Comment:
I'm pretty sure this is a bug, if everywhere it tries is impassable, tries is going to be set to 10, and then it's going to try to set a trap somewhere invalid, possibly triggering an assertion error.

Looking at it, it *also* looks like there was already a latent bug where it would only try to overwrite locations that already had a trap, which is the opposite of how it's supposed to work.

Reply:
What if rewrite it like so?
```cpp
    if( x_in_y( 3.0, 4.0 ) ) {
        tries = 0;
        bool found = false;
        do {
            goop.x = p->posx() + rng( -2, 2 );
            goop.y = p->posy() + rng( -2, 2 );
            tries++;
            found = g->m.passable( goop ) && g->m.tr_at( goop ).is_null();
        } while( !found && tries < 10 );
        if( found ) {
            if( g->u.sees( goop ) ) {
                add_msg( m_warning, _( "A nearby splatter of goo forms into a goo pit." ) );
            }
            g->m.trap_set( goop, tr_goo );
        } else {
            return 0;
        }
    }
```
Only one iteration.
Check if there is no trap.
`!one_in( 4 ) ==> x_in_y( 3.0, 4.0 ) `

Reply:
Yea that works, changes the logic a bit, but the original behavior where it can place multiple pits doesn't make sense.